src/BaseRepository.php

```diff
@@ -136,11 +136,13 @@ public function insert(array $fields)
 	{
 		$return = $this->dba->insert(static::$table_name, $fields);
 
-		if ($return) {
-			$fields['id'] = $this->dba->lastInsertId();
-			$return = $this->create($fields);
+		if (!$return) {
+			throw new HTTPException\InternalServerErrorException('Unable to insert new row in table "' . static::$table_name . '"');
 		}
 
+		$fields['id'] = $this->dba->lastInsertId();
+		$return = $this->create($fields);
```

Comment:
Here, you can use the "createFromPrototype" functionallity, so we wouldn't have to create a new instance (performance boost :-) )

Reply:
But we don't have a prototype at hand at this point?

Reply:
ahh .. yes you're right .. I oversaw that we're in `insert` and thought we're at `update` ... ^^